packages/block-editor/src/components/link-control/index.js

```diff
@@ -192,6 +201,14 @@ function LinkControl( {
 		isEndingEditWithFocus.current = false;
 	}, [ isEditingLink, isCreatingPage ] );
 
+	useEffect( () => {
+		dispatch( preferencesStore ).set(
+			'core/editor',
+			'linkControlSettingsOpen',
+			settingsOpen
+		);
+	}, [ settingsOpen ] );
+
```

Comment:
@getdave As per the comment you made on my earlier PR https://github.com/WordPress/gutenberg/pull/47930#discussion_r1104228612

I should not query the store in the component itself. So I might have to refactor the code.